lib/zinnia.js

```diff
@@ -133,11 +187,24 @@ export async function run ({
       console.error(`Zinnia closed all stdio with code ${code ?? '<no code>'}`)
       childProcess.stderr.removeAllListeners()
       childProcess.stdout.removeAllListeners()
+      if (shouldRestart) return
       const err = new Error(`Zinnia exited ${exitReason ?? 'for unknown reason'}`)
       maybeReportCrashToSentry(err)
       throw err
     })()
   ])
+
+  if (shouldRestart) {
+    return run({
```

Comment:
Should we wait until `childProcess` exits before we start another Zinnia instance?

Reply:
We are doing that, as the `Promise.all()` right above also waits for the child process and stdio to close

Reply:
Cool, thanks for double-checking 👍🏻